### PR TITLE
Use komega in suite tests

### LINKED ISSUE #208: Use `komega` in suite tests
**What would you like to be added**:

Use `komega` in the loadbalancer and node controller suite tests where applicable.

### PR DESCRIPTION
**Pull request by ushabelgur**
# Proposed Changes
- Use komega in tests-

Fixes #208

### CODE REVIEW

File: pkg/cloudprovider/onmetal/instances_v2_test.go [hunk 1]
```diff
@@ -101,40 +102,31 @@ var _ = Describe("InstancesV2", func() {
...
-						Type:    corev1.NodeExternalIP,
-						Address: "10.0.0.10",
-					},
-					{
-						Type:    corev1.NodeInternalIP,
-						Address: "10.0.0.1",
-					},
+		instanceMetadata, err := instances.InstanceMetadata(ctx, node)
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(instanceMetadata).Should(SatisfyAll(HaveField("ProviderID", fmt.Sprintf("%s://%s/%s", ProviderName, machine.Namespace, machine.Name)),
```
Comment (afritzler) [inline pkg/cloudprovider/onmetal/instances_v2_test.go:117]: Can you put the `HaveField` on a new line and factor out the `fmt.Sprintf` into a helper method.

File: pkg/cloudprovider/onmetal/instances_v2_test.go [hunk 2]
```diff
@@ -101,40 +102,31 @@ var _ = Describe("InstancesV2", func() {
...
+				},
+				corev1.NodeAddress{
+					Type:    corev1.NodeInternalIP,
+					Address: "10.0.0.1",
 				},
-				Zone:   "zone1",
-				Region: "",
-			}))
-		}).Should(Succeed())
+			)), HaveField("Zone", "zone1"), HaveField("Region", "")))
```
Comment (afritzler) [inline pkg/cloudprovider/onmetal/instances_v2_test.go:128]: Put each `HaveField` on a new line for better readability please.

File: pkg/cloudprovider/onmetal/load_balancer_test.go [hunk 3]
```diff
@@ -211,27 +206,31 @@ var _ = Describe("LoadBalancer", func() {
...
-				Ingress: []corev1.LoadBalancerIngress{{IP: "10.0.0.1"}},
-			}))
-		}).Should(Succeed())
+		lbStatus, err := olb.EnsureLoadBalancer(ctx, clusterName, service, []*corev1.Node{node1})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(lbStatus).To(Equal(&corev1.LoadBalancerStatus{
+			Ingress: []corev1.LoadBalancerIngress{{IP: "10.0.0.1"}},
+		}))
+
+		lbObject := &networkingv1alpha1.LoadBalancer{ObjectMeta: metav1.ObjectMeta{Name: lbName, Namespace: service.Namespace}}
```
Comment (afritzler) [inline pkg/cloudprovider/onmetal/load_balancer_test.go:223]: You can reuse the `lb` object defined in L209.

File: pkg/cloudprovider/onmetal/load_balancer_test.go [hunk 4]
```diff
@@ -261,29 +266,24 @@ var _ = Describe("LoadBalancer", func() {
...
+		Eventually(Object(lbRouting)).Should(SatisfyAll(
+			HaveField("ObjectMeta.OwnerReferences", ContainElement(metav1.OwnerReference{
 				APIVersion: "networking.api.onmetal.de/v1alpha1",
 				Kind:       "LoadBalancer",
 				Name:       lbName,
 				UID:        lb.UID,
-			}))
-			g.Expect(items).To(Equal(lbRouting.Destinations))
-		}).Should(Succeed())
+			})), HaveField("Destinations", items)))
```
Comment (afritzler) [inline pkg/cloudprovider/onmetal/load_balancer_test.go:278]: I would not collect the `items` above. Instead directly use an array here as we know how the `items` should look like. Please avoid if statements in test cases as there should be no conditioned execution in unit/suite tests.

File: pkg/cloudprovider/onmetal/load_balancer_test.go [hunk 5]
```diff
@@ -261,29 +266,24 @@ var _ = Describe("LoadBalancer", func() {
...
-		Eventually(func(g Gomega) {
-			err := olb.UpdateLoadBalancer(ctx, clusterName, service, []*corev1.Node{node2})
-			g.Expect(err).NotTo(HaveOccurred())
-			lbRouting := &networkingv1alpha1.LoadBalancerRouting{}
-			Expect(k8sClient.Get(ctx, types.NamespacedName{Namespace: service.Namespace, Name: lbName}, lbRouting)).To(Succeed())
-			g.Expect(items).To(Equal(lbRouting.Destinations))
-		}).Should(Succeed())
+		err = olb.UpdateLoadBalancer(ctx, clusterName, service, []*corev1.Node{node2})
+		Expect(err).NotTo(HaveOccurred())
+		lbRouting = &networkingv1alpha1.LoadBalancerRouting{ObjectMeta: metav1.ObjectMeta{Namespace: service.Namespace, Name: lbName}}
```
Comment (afritzler) [inline pkg/cloudprovider/onmetal/load_balancer_test.go:284]: Please use the routing object defined above here.

File: pkg/cloudprovider/onmetal/load_balancer_test.go [hunk 6]
```diff
@@ -261,29 +266,24 @@ var _ = Describe("LoadBalancer", func() {
...
+		Eventually(Object(lbRouting)).Should(SatisfyAll(
+			HaveField("Destinations", items)))
```
Comment (afritzler) [inline pkg/cloudprovider/onmetal/load_balancer_test.go:286]: Here again ensure on a concrete array which you define.

File: pkg/cloudprovider/onmetal/load_balancer_test.go [hunk 7]
```diff
@@ -312,22 +312,18 @@ var _ = Describe("LoadBalancer", func() {
...
-			ingresses := status.Ingress
-			for _, ingress := range ingresses {
-				g.Expect(ingress.IP).To(Equal(service.Status.LoadBalancer.Ingress[0].IP))
-			}
-		}).Should(Succeed())
+		status, exist, err := olb.GetLoadBalancer(ctx, clusterName, service)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(exist).To(BeTrue())
+		ingresses := status.Ingress
+		for _, ingress := range ingresses {
```
Comment (afritzler) [inline pkg/cloudprovider/onmetal/load_balancer_test.go:319]: Instead of using a for loop use a `ContainsElements` on the `Ingress` array in the status.

**Review by afritzler — CHANGES_REQUESTED** [review]
Thanks a lot for you PR. Please have a look at my comments.

File: pkg/cloudprovider/onmetal/instances_v2_test.go (hunk 2 above)
Comment (ushabelgur) [reply pkg/cloudprovider/onmetal/instances_v2_test.go]: Done

File: pkg/cloudprovider/onmetal/load_balancer_test.go (hunk 4 above)
Comment (ushabelgur) [reply pkg/cloudprovider/onmetal/load_balancer_test.go:278]: Done

File: pkg/cloudprovider/onmetal/load_balancer_test.go (hunk 5 above)
Comment (ushabelgur) [reply pkg/cloudprovider/onmetal/load_balancer_test.go:284]: Done

File: pkg/cloudprovider/onmetal/load_balancer_test.go (hunk 7 above)
Comment (ushabelgur) [reply pkg/cloudprovider/onmetal/load_balancer_test.go:319]: Modified test accordingly

File: pkg/cloudprovider/onmetal/load_balancer_test.go (hunk 6 above)
Comment (ushabelgur) [reply pkg/cloudprovider/onmetal/load_balancer_test.go:286]: done

File: pkg/cloudprovider/onmetal/load_balancer_test.go (hunk 5 above)
Comment (afritzler) [inline pkg/cloudprovider/onmetal/load_balancer_test.go:284]: Please remove this `lbRouting` and use the one defined in L271.

**Review by afritzler — CHANGES_REQUESTED** [review]

**Review by afritzler — APPROVED** [review]

**Pull request merged**